connect/api/src/main/java/org/apache/kafka/connect/data/Values.java

```diff
@@ -857,6 +857,7 @@ protected static SchemaAndValue parse(Parser parser, boolean embedded) throws No
         }
         if (firstCharIsDigit) {
             // Check for a date, time, or timestamp ...
+            token = token.replace("\\", "");
```

Comment:
Why are we making this change? No Connect literal value needs to have the colon's replaced in the ISO 8601 format, and the tests below appear to be trying to replicate how `Properties.store()` escape colons. But we use unescaped colons in many Kafka and Connect config files, and I don't think we should bother with trying to unnecessarily unescape certain characters.

Reply:
This line is a result of the development comment below, having to escape the literal value of a time. The length of the literal is checked in order to be parsed to a valid time, so I have to remove the escape characters.

Reply:
But why are there escape characters in the first place?

Reply:
I just explained it in the open discussion below.